Modules/_pickle.c

```diff
@@ -2063,30 +2065,35 @@ _Pickler_write_large_bytes(
     assert(self->write != NULL);
     PyObject *result;
 
-    // Commit the previous frame
-    if (_Pickler_CommitFrame(self, 0))
+    /* Commit the previous frame. */
+    if (_Pickler_CommitFrame(self, 0)) {
         return -1;
+    }
 
-    // Create a new frame dedicated to the large bytes
-    if (_Pickler_Write(self, header, header_size) < 0)
+    /* Create a new frame dedicated to the large bytes. */
+    if (_Pickler_Write(self, header, header_size) < 0) {
         return -1;
-    if (_Pickler_CommitFrame(self, payload_size))
+    }
+    if (_Pickler_CommitFrame(self, payload_size)) {
         return -1;
-
-    // Dump the buffer to the file
-    if (_Pickler_FlushToFile(self) < 0)
+    }
+    /* Dump the buffer to the file. */
+    if (_Pickler_FlushToFile(self) < 0) {
         return -1;
+    }
 
-    // Reinitialize the buffer for subsequent calls to _Pickler_Write.
-    if (_Pickler_ClearBuffer(self) < 0)
+    /* Reinitialize the buffer for subsequent calls to _Pickler_Write. */
+    if (_Pickler_ClearBuffer(self) < 0) {
         return -1;
+    }
 
-    // Stream write the payload into the file without going through the
-    // output buffer
+    /* Stream write the payload into the file without going through the
+       output buffer. */
     result = PyObject_CallFunctionObjArgs(self->write, payload, NULL);
     Py_XDECREF(result);
-    if (result == NULL)
+    if (result == NULL) {
```

Comment:
Comparing a pointer to a freed memory with NULL is an undefined behavior.

You don't need to decref it if it is NULL.